components/org.wso2.carbon.identity.api.user.recovery/org.wso2.carbon.identity.rest.api.user.recovery.v2/src/main/java/org/wso2/carbon/identity/rest/api/user/recovery/v2/impl/core/utils/RecoveryUtil.java

```diff
@@ -0,0 +1,576 @@
+import org.wso2.carbon.identity.recovery.IdentityRecoveryConstants;
+import org.wso2.carbon.identity.recovery.dto.NotificationChannelDTO;
+import org.wso2.carbon.identity.rest.api.user.recovery.v2.impl.core.Constants;
+import org.wso2.carbon.identity.rest.api.user.recovery.v2.impl.core.exceptions.ConflictException;
+import org.wso2.carbon.identity.rest.api.user.recovery.v2.impl.core.exceptions.ForbiddenException;
+import org.wso2.carbon.identity.rest.api.user.recovery.v2.impl.core.exceptions.InternalServerErrorException;
+import org.wso2.carbon.identity.rest.api.user.recovery.v2.impl.core.exceptions.NotAcceptableException;
+import org.wso2.carbon.identity.rest.api.user.recovery.v2.impl.core.exceptions.NotFoundException;
+import org.wso2.carbon.identity.rest.api.user.recovery.v2.impl.core.exceptions.PreconditionFailedException;
+
+import org.wso2.carbon.identity.rest.api.user.recovery.v2.model.APICall;
+import org.wso2.carbon.identity.rest.api.user.recovery.v2.model.ErrorResponse;
+import org.wso2.carbon.identity.rest.api.user.recovery.v2.model.Property;
+import org.wso2.carbon.identity.rest.api.user.recovery.v2.model.RecoveryChannel;
+import org.wso2.carbon.identity.rest.api.user.recovery.v2.model.RetryErrorResponse;
+import org.wso2.carbon.identity.rest.api.user.recovery.v2.model.UserClaim;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+
+import static org.wso2.carbon.identity.api.user.common.Constants.TENANT_CONTEXT_PATH_COMPONENT;
+import static org.wso2.carbon.identity.application.common.util.IdentityApplicationConstants.Error.UNEXPECTED_SERVER_ERROR;
+
+/**
+ * Contains the recovery endpoint utils.
+ */
+public class RecoveryUtil {
+
+    private static final Log log = LogFactory.getLog(RecoveryUtil.class);
+    private static final String LOG_MESSAGE_PREFIX = "INITIATOR";
+    private static final String FORBIDDEN_ERROR_CATEGORY = "FORBIDDEN_ERROR_CATEGORY";
+    private static final String CONFLICT_REQUEST_ERROR_CATEGORY = "CONFLICT_REQUEST_ERROR_CATEGORY";
+    private static final String REQUEST_NOT_FOUND_ERROR_CATEGORY = "REQUEST_NOT_FOUND_ERROR_CATEGORY";
+    private static final String REQUEST_NOT_ACCEPTABLE_ERROR_CATEGORY = "REQUEST_NOT_ACCEPTABLE_ERROR_CATEGORY";
+    private static final String RETRY_ERROR_CATEGORY = "RETRY_ERROR_CATEGORY";
+
+    // Map with the error codes categorized in to different error groups.
+    private static final HashMap<String, String> clientErrorMap = generateClientErrorMap();
+
+    /**
+     * Converts a list of UserClaim in to a UserClaim array.
+     *
+     * @param userClaimsList UserClaims List
+     * @return Map of user claims
+     */
+    public static HashMap<String, String> buildUserClaimsMap(List<UserClaim> userClaimsList) {
+
+        HashMap<String, String> userClaims = new HashMap<>();
+        for (UserClaim userClaimModel : userClaimsList) {
```

Comment:
Dont we need the null check for the `userClaimsList`

Reply:
It's not needed. If the `claims` property is not passed in the request, `userClaimsList` will be set as an empty list.